crates/nu-command/src/conversions/into/bool.rs

```diff
@@ -116,13 +116,13 @@ fn into_bool(
 }
 
 fn string_to_boolean(s: &str, span: Span) -> Result<bool, ShellError> {
-    match s.trim().to_lowercase().as_str() {
+    match s.trim().to_ascii_lowercase().as_str() {
         "true" => Ok(true),
         "false" => Ok(false),
         o => {
             let val = o.parse::<f64>();
             match val {
-                Ok(f) => Ok(f.abs() >= f64::EPSILON),
+                Ok(f) => Ok(f != 0.0),
```

Comment:
Comparing to `f64::EPSILON` is almost never what you actually want to do. Yes, when comparing floats, you want to do so with *an* epsilon because of floating point imprecision, but the machine epsilon is an astoundingly bad choice for a comparison epsilon. There is no "perfect" comparison epsilon, and the comparison epsilon really wants to be relative to the scale of the compared values, due to the floating nature of floating point arithmetic.

A comparison to machine epsilon is in most cases no different to a direct comparison with `0.0`, except for a false appearance of satisfying a rule of "no float `==`" just for the purpose of satisfying the rule. Equality with `0` is both the value it makes most sense to directly compare to ("did they write a literal zero") and more honest than comparing with machine epsilon.

Yes, the clippy lint suggested comparing with `f64::EPSILON` for a long time. This was unfortunately bad advice that took an unfortunately long period to replace, due to there not being any straightforward advice to replace it with. You essentially just can't check floats for "effective equality" without domain specific knowledge.

Reply:
Some recent rustlang discussion: https://github.com/rust-lang/rust/issues/116916

Reply:
And on clippy: https://github.com/rust-lang/rust-clippy/issues/6816

Reply:
Would you mind moving the float comparison into a separate PR? It is an issue on its own, it would be better addressed separately, IMO.